include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -157,6 +190,22 @@ namespace oneapi
 {
 namespace dpl
 {
+
+template <typename T>
+struct is_indirectly_device_accessible_iterator
+    : decltype(oneapi::dpl::__internal::__is_onedpl_indirectly_device_accessible_iterator(std::declval<T>()))
+{
+    static_assert(
+        std::is_same_v<decltype(decltype(oneapi::dpl::__internal::__is_onedpl_indirectly_device_accessible_iterator(
+                           std::declval<T>()))::value),
+                       const bool>,
```

Comment:
Should we really use here `const bool` or simple `bool` is ok too?

https://stackoverflow.com/questions/1443659/does-it-make-sense-to-return-fundamental-types-by-const-value
"
To be a little more specific, only "objects" can be const. The C++ standard's definition of "object" includes everything an lvalue refers to ("has a name") and class-type temporaries. A boolean return value is an rvalue of a non-class type which is why a standards-compliant compiler will just ignore "const" in this case. As others said already, it's useless in this context.
"

Reply:
Here we are attempting to detect if something is not a `bool_constant` as a return type from the ADL customization point.  
`bool_constant`-s are defined to be `integral_constant<bool, v>`, and those must have have a `constexpr bool value` member constant. `constexpr` variables are `const`, so when we are checking `is_same_v` we need to match or decay the `const` away.  Since we know that these should be `const bool` by the specification, we need to check against the same. I've tried this out in practice, and at least with icpx we need `const bool` or a `std::decay_t`, I believe any valid implementation would be the same. https://en.cppreference.com/w/cpp/types/integral_constant

Reply:
Understood your point, ok.